test/unit/main.js

```diff
@@ -1,3 +1,5 @@
+import './karma/karma-enforce-precompile';
```

Comment:
you have added `customHeaders` in karma.conf. why still need this?

Reply:
The custom headers will detect things like `eval` which we will not have as long as we are using `handlebars/runtime`. If we accidentally switch to full handlebars, the CSP will start failing again.

This check in karma detects a template that has not been precompiled. This would not throw a CSP error, instead the template would not be properly rendered. Depending on how the test is written we may not detect that or it may be confusing to the developer as to why the test is failing.

Reply:
Then shall the changes made to framework itself (`courage-for-signin-widget/src/`)?

Reply:
I did originally have this check in CourageForSigninWidget.js, but I decided to move it to karma because I didn't want a runtime penalty of looking for `{{` in every template string, and I also want to be careful about causing a runtime error visible to users. Seeing a UI with `{{` in it is a lower priority bug than having an exception which breaks the whole login flow. Of course the karma check only works if templates are rendered within unit tests.  Another option would be an eslint  plugin to detect strings with `{{` that are not wrapped in `hbs`.

Reply:
`eslint plugin` sounds better!

Reply:
I updated the eslint plugin on the courage side and am copying it over with the build from courage-for-signin-widget.